yarn-project/circuit-types/src/tx_effect.ts

```diff
@@ -0,0 +1,60 @@
+import { ContractData, PublicDataWrite, TxL2Logs } from '@aztec/circuit-types';
+import { Fr, MAX_NEW_COMMITMENTS_PER_TX } from '@aztec/circuits.js';
+
+export class TxEffect {
```

Comment:
Add a `hash` function to the `TxEffect` if you recall the calldata on `L2Block` it is just using the values from in here, so should be possible to just do it in here. Would also simplify that computation quite a lot.